Source/CParse/cscanner.c

```diff
@@ -350,24 +350,33 @@ static int yylook(void) {
 	}
       }
       break;
-      
+
     case SWIG_TOKEN_ELLIPSIS:
       return ELLIPSIS;
 
+    case SWIG_TOKEN_LLBRACKET:
+      do {
+        tok = Scanner_token(scan);
+      } while (tok != SWIG_TOKEN_RRBRACKET);
```

Comment:
This needs to handle `Scanner_token(scan)` returning `0` otherwise it will infinitely loop if the matching `]]` is missing.

There are some existing similar loops, but they're all skipping zero or more of a certain token so check for equality not inequality, and so implicitly end if `tok` is 0, e.g.:

```
          do {
            tok = Scanner_token(scan);
          } while (tok == SWIG_TOKEN_ENDLINE);
```